tensorflow/contrib/learn/python/learn/estimators/model_fn.py

```diff
@@ -127,7 +126,7 @@ def __new__(cls,
     ModeKeys.validate(mode)
 
     # Assert all ops are from the same graph.
-    get_graph_from_inputs((predictions, loss, train_op))
+    ops.IndexedSlices(predictions, loss, train_op)
```

Comment:
The contrib framework function exposes an internal function from `tensorflow.python.framework.ops`. Rather than do that, I rather directly call on the class from tensorflow.python.framework.ops which is `ops.IndexedSlices` which then calls its own internal function without jeopardizing protected access.

Reply:
It's a suprising usage of IndexedSlices. For example you're sending train_op as a dense_shape. I would revert this line. Otherwise LGTM